src/components/tl/ucp/tl_ucp_coll.c

```diff
@@ -51,6 +53,114 @@ static ucc_status_t ucc_tl_ucp_coll_finalize(ucc_coll_task_t *coll_task)
     return UCC_OK;
 }
 
+static ucc_status_t ucc_tl_ucp_triggered_coll_complete(ucc_coll_task_t *parent_task,
+                                                       ucc_coll_task_t *coll_task)
+{
+    ucc_tl_ucp_task_t *task = ucc_derived_of(coll_task, ucc_tl_ucp_task_t);
+
+    tl_info(task->team->super.super.context->lib,
+        "triggered collective complete. task:%p", coll_task);
+    return ucc_mc_ee_task_end(coll_task->ee_task, coll_task->ee->ee_type);
+}
+
+static ucc_status_t ucc_tl_ucp_event_trigger_complete(ucc_coll_task_t *parent_task,
+                                                      ucc_coll_task_t *coll_task)
+{
+    ucc_tl_ucp_task_t *task = ucc_derived_of(coll_task, ucc_tl_ucp_task_t);
+    ucc_status_t status;
+
+    tl_info(task->team->super.super.context->lib, "event triggered. task:%p", coll_task);
+
+    coll_task->ee_task = parent_task->ee_task;
+    status = coll_task->post(coll_task);
+    if (status == UCC_OK) {
```

Comment:
post return code does not mean completion. it only reports if error happend. if you need to check completion, you should check task status.

Reply:
fixed